js/entity_browser.multi_step_display.js

```diff
@@ -53,4 +77,123 @@
     }
   };
 
+  Drupal.entityBrowserMultiStepDisplay.removeEntity = function (entities_list_element, entity_id) {
+    var entities_list = $(entities_list_element);
+
+    // Add proxy element that will be replaced with returned Ajax Command.
+    var element_selector = '[data-drupal-selector="edit-selected-'.concat(entity_id.replace(/_/g, '-'), '"]');
+    entities_list.find(element_selector).remove();
+
+    if (!commands_queue['remove']) {
+      commands_queue['remove'] = [];
+    }
+
+    commands_queue['remove'].push(
+      {
+        entity_id: entity_id
+      }
+    );
+
+    Drupal.entityBrowserMultiStepDisplay.loadQueuedEntities(entities_list_element, typeof entity_id !== 'undefined');
+  };
+
+  /**
+   * Handle load entity over Ajax Request in selection display.
+   *
+   * @param {object} event
+   *   Event object.
+   * @param {Array} entity_ids
+   *   Entity Id that should be loaded into selection list.
+   */
+  Drupal.entityBrowserMultiStepDisplay.loadEntities = function (event, entity_ids) {
```

Comment:
Agreed - I would prefer it each Widget broadcast JS events for adding and removing Entities, and this code simply waited for those events to happen and made AJAX calls when they did.